pumpkin-solver/src/engine/cp/propagation/propagator.rs

```diff
@@ -177,7 +178,7 @@ pub trait Propagator {
     /// Logs statistics of the propagator using the provided [`StatisticLogger`].
     ///
     /// It is recommended to create a struct through the [`create_statistics_struct!`] macro!
-    fn log_statistics(&self, _statistic_logger: StatisticLogger) {}
+    fn write_statistics(&self, _writer: &mut Box<dyn Write>, _statistic_logger: StatisticLogger) {}
```

Comment:
The documentation for this function should probably be updated based on the additional argument